### DISCUSSION TITLE
SHAP values on panel data modeled by sklearn's GradientBoostingRegressor

### DISCUSSION THREAD

**Discussion by jedsd3285** [shap/shap#2112] [Show and tell]
I have annual panel data of numerical features for particular geographic regions across 5 years (so two dimensions: entity, and time). Unfortunately I could not get Sigrist's GPBoost to work :( and so I did the unthinkable and regressed the whole panel dataset using sklearn's GBR just to get started, all the while using the assumptions/pitfalls mentioned by Sigrist that come along since I'm using panel (ie that the target features are not related though they are, discontinuous approach by GBR for entities.. etc. etc.). I did not even use a cross validation set to toggle hyperparameters and explicitly overtrained on the whole dataset. (I said what the heck .. let's see how this turns out.)

Curiously enough though, I used SHAP to generate dependence plots and saw very interesting relationships in some cases. I am loathe to publish those of course knowing the underlying approach of the model is somewhat (mostly?) .. . not ideal. However, would SHAP somehow overcome limitations of using plain-vanilla GBR on panel data and thus show insights still of the data? What valid interpretations to feature interactions could even be made with this? Is it equivalent to just repeating the same relationships over and over since if the time and entity columns are not represented as training features? My thought was, well fine GBR overtrains on the data, and I want to let the data speak. It has spoken, but in a foreign language I'm unsure how to translate.

Sample image: the effects of one feature 'NumOffices' seems to add to a step-wise function to another feature 'Tot Popln'. Implications are interesting if this is reliable, since the target feature are annual bank deposit levels by geographic areas over about 5 years. Again though, the geographic area and the year represented are omitted, and raw level data is just treated as one whole dataset.

![sample jsd](https://user-images.githubusercontent.com/87921090/127242385-c4cfcd9c-976a-4c43-8505-6431f0a85fa1.png)

TY :)

PS - I am in love with SHAP values ..  just discovered them this past weekend! excellent!
